commands/bash/host/config/config

```diff
@@ -12,19 +12,22 @@ function config() {
     fi
     case "$subcommand" in
         get)
-            result="$(jq -r ".$1" "$config_path")"
+            default="${2:-}"
+            result="$(jq -r --arg key "$1" 'getpath($key | split("."))' "$config_path")"
             if [[ ! "$result" == null ]]; then
                 echo $result
+            elif test -n "$default"; then
+                echo $default
             fi
         ;;
         set)
-            jq -c --arg key "$1" --arg val "$2" '. + { ($key): $val }' "$config_path" > "tmp.$$.json" && mv "tmp.$$.json" "$config_path"
+            jq --arg key "$1" --arg val "$2" 'setpath($key | split("."); $val)' "$config_path" > "tmp.$$.json" && mv "tmp.$$.json" "$config_path"
         ;;
         view)
             cat "$config_path"
         ;;
         unset)
-            jq -c "del(.$1)" "$config_path" > "tmp.$$.json" && mv "tmp.$$.json" "$config_path"
+            jq --arg key "$1" 'delpaths([$key | split(".")])' "$config_path" > "tmp.$$.json" && mv "tmp.$$.json" "$config_path"
```

Comment:
some context/reasoning around these changes would be greatly appreciated :)

Reply:
Basically, we were passing in the key/value as --args to jq during set, but not during get or unset.  As a result, we got different behavior between those two cases.  In set, a dotted key (e.g., minikube.vm_driver) would be set literally --- `"minikube.vm_driver": "virtualbox"` --- but in unset and get, it would look for 
```
"minikube": {
    "vm_driver": "virtualbox"
}
```

Unfortunately, using --arg is safer than directly interpolating the key/value into the string (reduces confusion around what is parsing '$'; prevents injection).

So, to recap:

for get/unset, we were interpolating dotted values directly into the query string, and getting e.g. .minikube.vm_driver.

For set, we were passing in the key as an argument, but then using it wrong (effectively doing .["minikube.vm_driver"]).

This new code always passes values in as arguments, and then uses them correctly from within jq to address their corresponding locations in the hierarchy.